python/dgl/graphbolt/base.py

```diff
@@ -102,6 +103,33 @@ def expand_indptr(indptr, dtype=None, node_ids=None, output_size=None):
     )
 
 
+def index_select(tensor, index):
```

Comment:
Does this support tensor on CUDA as well?

Reply:
Yes, it supports CPU, Pinned-GPU and GPU cases all.